server/app/filters/CiviFormProfileFilter.java

```diff
@@ -51,6 +53,8 @@ private boolean shouldRedirect(Http.RequestHeader requestHeader) {
   public CompletionStage<Result> apply(
       Function<Http.RequestHeader, CompletionStage<Result>> nextFilter,
       Http.RequestHeader requestHeader) {
+    System.out.println("should redirect in ProfileFilter..." + shouldRedirect(requestHeader));
```

Comment:
Left these print statements in because they were helpful to me to understand when a guest session was being created.